### DISCUSSION TITLE
[Chapter 09] `Netty4ClientHttpRequestFactory`와 비동기 논블로킹 프레임 워크에 대한 고찰

### DISCUSSION THREAD

**Discussion by Irisation23** [cowaine/spring-boot-for-msa#140] [Chapter 09. Application Integration: REST API] [answered]
비동기, 논블로킹이란 논제는 개발자의 방향에 있어서 한발 더 진보할 수 있는 방향성을 제시할 수 있는 개념이라는 생각이 됩니다.

그리고 9.1.1을 읽다 보면 `ClientHttpRequestFacotry` 인터페이스의 구현체 중 하나인 `Netty4ClientHttpRequestFactory` 가 등장하며 설명은 다음과 같습니다.

```
비동기 논블로킹 프레임워키인 Netty를 사용하여 작성된 구현체다.
비동기 프로그래밍을 계획한다면 가장 먼저 고려하는 것이 좋다.
```

저는 해당 구문을 보며 다음과 같은 생각을 하게 되었습니다.
비동기 논블로킹 역시도 해당 책에서 제시하는 답변의 대부분을 차지하는 `상황에 알맞게 사용하라.` 라는 모토에서 어긋나지 않으리라 생각이 됩니다.

그리고 궁금한 점은 다음과 같습니다.

1. 비동기 논블로킹으로 완벽하게 구현된 서비스는 어떤 것이 있는지?
2. 만약 그렇다면 동기 블로킹 보다 어떤 점이 좋은지?

감사합니다.

**Comment by JoisFe** [comment]
@KIrisation 님께서 질문하신 내용과 정말 적합한 영상인 것 같아 링크 남깁니다. (1번 질문은 전혀 모르겠네요.. ㅎㅎ)

- [참고 영상](https://www.youtube.com/watch?v=4x1QRyMIjGU)

**Comment by corock** [comment, marked as answer]
- [사용하면서 알게 된 Reactor, 예제 코드로 살펴보기](https://tech.kakao.com/2018/05/29/reactor-programming)
- [스프링캠프 2017 [Day1 A3] : Spring Web Flux](https://www.youtube.com/watch?v=2E_1yb8iLKk)
- [스프링5 웹플럭스와 테스트 전략](https://tv.kakao.com/channel/3150758/cliplink/391418995)
- [3월 우아한 Tech 세미나 후기](https://techblog.woowahan.com/2619)

**Comment by Irisation23** [comment]
### 금주 Check Point

- Node.js가 비동기 논블로킹을 사용하고 있음
